src/settings/Settings.ts

```diff
@@ -768,4 +775,307 @@ export const SETTINGS: {[setting: string]: ISetting} = {
         supportedLevels: LEVELS_UI_FEATURE,
         default: true,
     },
+    // TODO: TravisR will fix this nightmare
```

Comment:
What is @turt2live going to work on here?
I'm slightly concerned looking at the implementation here that keybindings will go out of sync with the categories.

Ideally `ISetting` would have an extra parameter which would set the category. That way it would be possible in the keybindings view to iterate through the enumeration of category of filter through the relevant settings

It would help if I understand the scope of the work that is planned to see whether my concerns are still valid or not

Reply:
This is a reminder I asked for as part of https://github.com/matrix-org/matrix-react-sdk/pull/5669

Reply:
> It would help if I understand the scope of the work that is planned to see whether my concerns are still valid or not

Sure, I probably should have included more details in the description.

The goal is to add customisable keybindings. A [PR to decouple key bindings from event handling](https://github.com/matrix-org/matrix-react-sdk/pull/5720) has already been merged. Each `KeyBinding.Name` setting in `SETTINGS` will represent one keybinding. We can add a new `IKeyBindingsProvider` that will get the keybindings from the settings. It will be enabled only if the labs flag is - this will allow us to keep default keybindings untouched while we play with customizable keybindings in labs. The `KeybindingUserSettingsTab` can then be modified to support changing the keybindings. 

If you have any questions be sure to ask in a DM (or in #element-web - opinions of other people might be nice).

> Ideally ISetting would have an extra parameter which would set the category. That way it would be possible in the keybindings view to iterate through the enumeration of category of filter through the relevant settings

Originally, I wanted to do something like `Keybinding.Category.NameOfKeybinding` but after [a conversation with @turt2live](https://matrix.to/#/!bEWtlqtDwCLFIAKAcv:matrix.org/$L29CzoLkqn5N9Kzp1i8M6B5yteXScc5-JwvWemyaz1Q?via=matrix.org&via=vector.modular.im&via=chat.weho.st), the decision was to just use one layer to identify that it is a keybinding because multiple-layer would be hard to manage